emva1288/unittests/test_results.py

```diff
@@ -300,3 +302,66 @@ def test_nans(self):
         self.assertIs(r.DSNU1288_DN(), np.nan)
 
         del r
+
+
+class TestRoutines(unittest.TestCase):
+
+    def setUp(self):
+        # instantiate a bayer filter and a wrong filter
+        bayer_filter = get_bayer_filter(0, 1, 1, 0, 1000, 1000)
+        wrong_filter = np.tile([[1, 1, 1, 0], [1, 0, 0, 1]], (500, 250))
+
+        # Source images. imgc is for color image (image with a bayer filter)
+        self.img = np.abs(np.random.random([1000, 1000]))
+        self.imgc = np.ma.array(self.img, mask=bayer_filter)
+
+        # Image wrong's purpose is to assert that a ValueError is raised
+        self.img_wrong = np.ma.array(self.img, mask=wrong_filter)
+
+        # Pass the highpass filter on the source images
+        res = high_pass_filter(self.img, 5)
+        self.img_filt = res['img']/res['multiplicator']
+        res = high_pass_filter(self.imgc, 5)
+        self.imgc_filt = res['img']/res['multiplicator']
+
+        # To assert that the highpass filter does its job, a low frequency
+        # signal will be introduced in the image
+        signal = np.sin(np.linspace(0, np.pi, 1000))
+        sig_img = [line + val for line, val in zip(self.img, signal)]
+        sig_imgc = np.ma.array(sig_img, mask=bayer_filter)
+
+        # filtering the images with a low frequency signal
+        res = high_pass_filter(sig_img, 5)
+        self.sig_img_filt = res['img']/res['multiplicator']
+        res = high_pass_filter(sig_imgc, 5)
+        self.sig_imgc_filt = res['img']/res['multiplicator']
+
+    def test_highpass_filter(self):
+        # Assert raises in case of bad kernel size or bad mask
+        with self.assertRaises(ValueError):
+            high_pass_filter(self.img_wrong, 5)
+        with self.assertRaises(ValueError):
+            high_pass_filter(self.imgc, 6)
+
+        # assert that the means of the filtered images are almost 0
+        self.assertAlmostEqual(0, np.mean(self.imgc_filt), 3)
+        self.assertAlmostEqual(0, np.mean(self.img_filt), 3)
+
+        # Assert that the std stays about the same after filtering an image
+        # without a low frequency signal
+        self.assertAlmostEqual(np.std(self.img),
```

Comment:
why the std should remain the same? A low frequency noise might change the std dramatically

Reply:
The idea here was to assert that in the case of an image without a low frequency signal, the filter does not affect the std. But after considerations its probably not necessary and np.random could generate an image with a low frequency. It has been removed